src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LinkedEditingRange/LinkedEditingRangeEndpoint.cs

```diff
@@ -93,17 +98,25 @@ public TextDocumentIdentifier GetTextDocumentIdentifier(LinkedEditingRangeParams
         _logger.LogInformation("LinkedEditingRange request was null at {location} for {uri}", location, request.TextDocument.Uri);
         return null;
 
-        static async Task<SourceLocation> GetSourceLocation(
+        async Task<SourceLocation?> GetSourceLocation(
             LinkedEditingRangeParams request,
             DocumentContext documentContext,
             CancellationToken cancellationToken)
         {
             var sourceText = await documentContext.GetSourceTextAsync(cancellationToken).ConfigureAwait(false);
             var linePosition = new LinePosition(request.Position.Line, request.Position.Character);
-            var hostDocumentIndex = sourceText.Lines.GetPosition(linePosition);
-            var location = new SourceLocation(hostDocumentIndex, request.Position.Line, request.Position.Character);
 
-            return location;
+            if (sourceText.IsLinePositionValid(linePosition, _logger))
+            {
+                var hostDocumentIndex = sourceText.Lines.GetPosition(linePosition);
+                var location = new SourceLocation(hostDocumentIndex, request.Position.Line, request.Position.Character);
+
+                return location;
+            }
+            else
+            {
+                return null;
+            }
```

Comment:
And again, this could be `request.Position.TryGetAbsoluteIndex(...)`

Reply:
Agreed, same comments as above.